rclpy/src/rclpy/serialization.cpp

```diff
@@ -0,0 +1,149 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Include pybind11 before rclpy_common/handle.h includes Python.h
+#include <pybind11/pybind11.h>
+
+#include <rcl/error_handling.h>
+#include <rcl/rcl.h>
+#include <rcl/types.h>
+#include <rcutils/allocator.h>
+#include <rcutils/error_handling.h>
+
+#include <memory>
+#include <stdexcept>
+
+extern "C"
+{
+#include "rclpy_common/common.h"
+}
+#include "rclpy_common/handle.h"
+
+#include "rclpy_common/exceptions.hpp"
+
+#include "serialization.hpp"
+
+namespace rclpy
+{
+class SerializedMessage
```

Comment:
@sloretz meta: perhaps a `struct` is more adequate?

Reply:
Made it a struct in 5fc0e07